src/main/java/bio/terra/service/snapshot/Snapshot.java

```diff
@@ -81,6 +81,11 @@ public Snapshot snapshotSources(List<SnapshotSource> snapshotSources) {
         return this;
     }
 
+    // TODO: When we support more than one dataset per snapshot, all uses of this will need refactoring.
+    public SnapshotSource getFirstSnapshotSource() {
+        return snapshotSources.get(0);
```

Comment:
Even though it likely can't happen, you should add a check to make sure that there is at least one element to avoid an array out of bounds exception (allows us to throw a more meaningful exception)